auth/credentials/detect.go

```diff
@@ -93,6 +93,7 @@ func DetectDefault(opts *DetectOptions) (*auth.Credentials, error) {
 			ProjectIDProvider: auth.CredentialsPropertyFunc(func(context.Context) (string, error) {
 				return metadata.ProjectID()
 			}),
+			UniverseDomainProvider: &internal.ComputeUniverseDomainProvider{},
```

Comment:
nit: maybe the struct returned by the computeTokenProvider method could have a method that impl's this and/or return a function to impl it. Then all the compute logic is tied to a single struct.

Reply:
after reviewing a bit more maybe this is fine as it is used in more than one spot, I will leave the comment open though for your consideration.

Reply:
Agreed, let's leave as-is.